app/src/main/java/com/aaronjwood/portauthority/async/ScanHostsAsyncTask.java

```diff
@@ -37,26 +37,26 @@ public ScanHostsAsyncTask(MainAsyncResponse delegate) {
      * @param params IP address
      */
     @Override
-    protected Void doInBackground(String... params) {
-        String ip = params[0];
-        String parts[] = ip.split("\\.");
+    protected Void doInBackground(Integer... params) {
+        int ipv4 = params[0];
+        int cidr = params[1];
 
         ExecutorService executor = Executors.newCachedThreadPool();
 
+        double hostBits = 32.0d - cidr; // How many bits do we have for the hosts.
+        int netmask = (0xffffffff >> (32 - cidr)) << (32 - cidr); // How many bits for the netmask.
+        int numberOfHosts = (int) Math.pow(2.0d, hostBits) - 2; // 2 ^ hostbits = number of hosts in integer.
```

Comment:
The `- 2` won't exclude the gateway, right? Isn't it some special address for the subnet itself?